### PR TITLE
Start on use-case oriented documentation

### PR DESCRIPTION
**Pull request by eliasson**
What about adding another type of documentation, one that is use-case oriented?

E.g. show how to assert when you have a list. Give tips and tricks both regarding the _Intent API_ as well as _practices_ on writing good tests (regardless of framework)?

I think such a document would serve new users as well as developers new to TDD / testing well. Intention is to showcase common situations and let the API specification (or matchers.md) cover the details.

Related to #27

### CODE REVIEW

**Comment by eliasson** [comment]
@provegard any thoughts on this style of documenting?

File: docs/how-to-assert.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+"using custom precision" in:
+  given customPrecision: intent.core.FloatingPointPrecision[Double]
+    def numberOfDecimals: Int = 2
+  expect(1.123456789d).toEqual(1.12d)
+```
+
+
+## A sequence
+
+Asserting some condition for a sequence is a very common scenarion.
```
Comment (provegard) [inline docs/how-to-assert.md]:
```suggestion
Asserting some condition for a sequence is a very common scenario.
```

File: docs/how-to-assert.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+[info] [FAILED] intent.docs.HowToAssert >> How to assert >> a sequence >> for a single element (8 ms)
+[info] 	Expected List(1, 2, 3) to contain 4 (intent/src/test/scala/intent/docs/HowToAssert.scala:31:21)
+```
+
+**TIP:**
+> Avoid matching a collection on its length, as a failure will not give any details on the element(s) that were
+> missing or extra. An alternative is to filter a larger list and match on a smaller inline list
+> ```scala
+> Example?
```
Comment (provegard) [inline docs/how-to-assert.md]: Difficult if you're after the length...

**Comment by provegard** [comment]
Very nice!

**Comment by eliasson** [comment]
I'll merge this to master and we'll keep it up-to-date there on common scenarios.

**Pull request merged**